File: LeetCode/Medium/0443-string-compression/0443-string-compression.cpp
class Solution {
public:
    int compress(vector<char>& chars) {
        char prev;
        int prev_count;
        int i = 0;
        int idx = 0;
        
        while (i < chars.size()) {
            prev = chars[i];
            prev_count = 1;
            ++i;
            while (i < chars.size() && prev == chars[i]) {
                ++i;
                ++prev_count;
            }
            
            chars[idx++] = prev;
            if (prev_count != 1) {
                for (char v : to_string(prev_count)) chars[idx++] = v;
            }
        }
        
        return idx;
    }
};

File: LeetCode/Medium/0443-string-compression/0443-string-compression.py
class Solution:
    def compress(self, chars: List[str]) -> int:
        prev = chars[0]
        prev_count = 1
        idx = 0
        
        for v in chars[1:]:
            if v == prev:
                prev_count += 1
            else:
                chars[idx] = prev
                idx += 1
                if prev_count != 1:
                    for num in str(prev_count):
                        chars[idx] = num
                        idx += 1
                prev = v
                prev_count = 1

        chars[idx] = prev
        idx += 1

        if prev_count != 1:
            for v in str(prev_count):
                chars[idx] = v
                idx += 1

        return idx
        